### PR TITLE
Align with facebook's RN on react-native bundle --dev for BundleCommand

### PR DESCRIPTION
**Pull request by namrog84**
`react-native bundle --dev` currently defaults to true
You can see for yourself by running `npx react-native bundle --help`

With this being set to true, this is what causes yellow box warnings appearing in release builds. Or at least it did at one point. Not sure what other effects it has. 

in Facebook's ReactNative android example build.
As shown here
https://github.com/facebook/react-native/blob/1fb815ed4314c1396999e64354ca0b5d4ed83663/react.gradle#L156

This PR lets you control it as well via a flag.  I am adding it so that all release builds will set it appropriately. But still allow dev debug bundles to work properly.

See more info at issue #4276 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4277)

### CODE REVIEW

File: packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj [hunk 1]
```diff
@@ -63,6 +63,12 @@
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <LinkIncremental>false</LinkIncremental>
   </PropertyGroup>
+  <PropertyGroup>
+    <devEnabled>true</devEnabled>
```
Comment (vmoroz) [inline packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj]: Please consider to add a prefix for the variable to disambiguate with any other MSBuild variables. This name looks too generic to be accidentally used by some other MSBuild task. E.g. it could be ReactNativeDevMode.
Also, it would be nice to use Pascal casing as other property names in this file.

File: packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj [hunk 2]
```diff
@@ -63,6 +63,12 @@
...
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)' == 'Release'">
```
Comment (vmoroz) [inline packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj]: Ideally it should be a property that we could override from the VS property sheet UI. It may be a subject for another PR though.  The experience must be similar to how we set the C++/WinRT properties. I did not do it myself, so I do not know all details. This articles seems to have some info about it: https://www.codeproject.com/articles/881782/creating-custom-ui-property-pages-sheets-in-visual

File: packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj (hunk 1 above)
Comment (acoates-ms) [reply packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj]: I agree something like JSBundleDevMode or something, since its only affecting if the JS bundle. Having said that, the same setting in RN android's gradle builds is just called devEnabled, so there is some precedence to the devEnable name. -- So I'm fine with this too.

Comment (acoates-ms) [reply packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj]: Actually thats just a local var.  The actual external gradle interface is more complex:
``` 
           // Set up dev mode
            def devEnabled = !(config."devDisabledIn${targetName}"
                || targetName.toLowerCase().contains("release"))```

File: packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj (hunk 2 above)
Comment (acoates-ms) [reply packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj]: Lets file an issue to get the proper property sheet UI, so we dont forget about it as an idea.

File: packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj (hunk 1 above)
Comment (namrog84) [reply packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj]: Updated

File: packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj (hunk 2 above)
Comment (namrog84) [reply packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj]: opened another issue to follow up on this
https://github.com/microsoft/react-native-windows/issues/4280

**Comment by namrog84** [comment]
Since it looks things got reworked into bundle-cpp or something to the package.json, I don't know how to pass this along properly as it's just not my area of knowledge and as the fix is already in our own project. 

I am just going to abandon this PR, and leave the issue #4276 open for whoever knows more about this this properly.
